pkg/tsdb/schema/schema.go

```diff
@@ -1,40 +1,66 @@
 package schema
 
 import (
+	"fmt"
 	"github.com/pkg/errors"
 	"github.com/v3io/v3io-tsdb/pkg/aggregate"
 	"github.com/v3io/v3io-tsdb/pkg/config"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"strconv"
 	"strings"
+	"time"
 )
 
 const (
-	SchemaVersion                 = 0
+	DefaultMaximumSampleSize    = 8       // bytes
+	DefaultMaximumPartitionSize = 1700000 // 1.7MB
+	DefaultMinimumChunkSize     = 200     // bytes
+	DefaultMaximumChunkSize     = 32000   // bytes
+
+	Version                       = 0
 	DefaultStorageClass           = "local"
 	DefaultIngestionRate          = ""
 	DefaultAggregates             = "" // no aggregates by default
 	DefaultAggregationGranularity = "1h"
 	DefaultShardingBuckets        = 8
 	DefaultLayerRetentionTime     = "1y"
 	DefaultSampleRetentionTime    = 0
-	DefaultPartitionInterval      = "48h"
-	DefaultChunkInterval          = "1h"
 )
 
-func NewSchema(aggregates []string) (*config.Schema, error) {
+func NewSchema(sampleRate, aggregatorGranularity, aggregatesList string, minChunkSize, maxChunkSize, maxSampleSize, maxPartitionSize, sampleRetention, shardingBuckets int) (*config.Schema, error) {
```

Comment:
I wouldn't expose the parameters for the rate calculation to the user. the way to change them is by config if you want. (`minChunkSize, maxChunkSize, maxSampleSize, maxPartitionSize`)